src/domain/backbone/backbone.repository.ts

```diff
@@ -1,16 +1,36 @@
 import { Inject, Injectable } from '@nestjs/common';
+import { DefinedError, ValidateFunction } from 'ajv';
+import { JsonSchemaService } from '../../common/schemas/json-schema.service';
+import { ValidationErrorFactory } from '../errors/validation-error-factory';
+import { backboneSchema } from '../balances/entities/schemas/backbone.schema';
 import { ITransactionApiManager } from '../interfaces/transaction-api.manager.interface';
 import { Backbone } from './entities/backbone.entity';
 
 @Injectable()
 export class BackboneRepository {
+  private readonly isValidBackbone: ValidateFunction<Backbone>;
+
   constructor(
     @Inject(ITransactionApiManager)
     private readonly transactionApiManager: ITransactionApiManager,
-  ) {}
+    private readonly validationErrorFactory: ValidationErrorFactory,
+    private readonly jsonSchemaService: JsonSchemaService,
+  ) {
+    this.isValidBackbone = this.jsonSchemaService.compile(
+      backboneSchema,
+    ) as ValidateFunction<Backbone>;
+  }
 
   async getBackbone(chainId: string): Promise<Backbone> {
     const api = await this.transactionApiManager.getTransactionApi(chainId);
-    return api.getBackbone();
+    const backbone = await api.getBackbone();
+
+    if (!this.isValidBackbone(backbone)) {
+      // TODO: probably we want to invalidate cache at this point
```

Comment:
This can now be more tricky. Since at this level you wouldn't know that there's a `cache`.

Reply:
I agree, in fact I removed these comments in my other PR for the same reason. Removing it here as well! 🙂

Fixed in [d376ff0](https://github.com/5afe/safe-client-gateway-nest/pull/82/commits/d376ff082174dc6624e1eeb6ec396f4bfbc36045)

Reply:
It's definitely something that we should consider. The invalidation can still happen on the `datasources` side but triggering it from `domain` needs some adjustment.

One thing that we need to implement are the invalidation hooks (route level) – so at some point you'd be able to invalidate the cache from `routes`